### PR TITLE
[O11y][Traefik] Add dimension fields for health datastream

### PR DESCRIPTION
**Pull request by harnish-elastic**
- Enhancement

## What does this PR do?

This PR adds dimension fields for health data stream to support TSDB enablement.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

- Relates #7766

## Screenshot

Refer: https://github.com/elastic/integrations/issues/7766#:~:text=Screenshots-,Health,-%7B%0A%20%20%20%20%22traefik%22%3A%20%7B%0A%20%20%20%20%20%20%20%20%22properties%22%3A%20%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20%22health

### CODE REVIEW

**Comment by agithomas** [comment]
If there are multiple "status_codes", will there be separate documents or just one document ?

**Comment by harnish-elastic** [comment]
> If there are multiple "status_codes", will there be separate documents or just one document ?

@agithomas Please refer to [this](https://doc.traefik.io/traefik/v1.5/configuration/backends/web/#health:~:text=%22total_status_code_count%22) response, the response will ingest to elasticsearch in one document because the field type is the object.

File: packages/traefik/changelog.yml [hunk 1]
```diff
@@ -1,3 +1,8 @@
+- version: "1.9.0"
+  changes:
+    - description: Add dimension fields to the health datastream to support TSDB.
```
Comment (ritalwar) [inline packages/traefik/changelog.yml:3]:
```suggestion
    - description: Add dimension mapping for `health` datastream.
```

**Review by ritalwar — APPROVED** [review]
LGTM!

**Review by agithomas — APPROVED** [review]
LGTM!

**Review by milan-elastic — APPROVED** [review]
LGTM!

**Pull request merged**